### DISCUSSION TITLE
Tracking Other Replaceables

### DISCUSSION THREAD

**Discussion by systematicallyrandom** [andrew-codechimp/HA-Battery-Notes#1220] [Ideas]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aissue+label%3Aenhancement).

### Is your feature request related to a problem? Please describe.

I would love to be able to track preventative maintenance tasks other than replacing batteries. For instance my furnace filter, water filters, or when my sceptic system was last pumped. Battery Notes already has most the underlying functionality already (date last changed, battery replaced button, battery type, etc.) and it seems like it could be easily extended to include other random trackables. The main change needed would be that there is not necessarily another existing entity to attach to like there is a battery (like my water filter has no related entity), so a new entity would have to be created vs extending an existing one. 

And probably the next logical change would be a way to specify the expected life/change frequency of the trackable since there is no battery percentage (or filter age) being reported for my water filter.

### Describe the solution you'd like

The ability to add "battery notes" not attached to an existing entity

A way to specify the trackable type (battery, filter, etc.)

A way to specify change frequency manually (ex. I want to replace my filter every 3 months). Along with this, the due date of the next change. Also, the templating function could be used (like the low battery status template) to track, for example, a water filter change after XXXX gallon/liters if water usage was available in HA.

### Describe alternatives you've considered

I've looked at Grocy as an alternative, but setup is complex, clunky to use, not native HA UI (requires an addon) and I can't easily use NFC tags or QR codes to trigger replacement events. (With Battery Notes, you can scan a tag which would trigger the battery replaced button.)

### Additional context

I realize the name of the integration would not be as applicable, so that is a consideration. Should this be a new integration or an extension of this one? It seems like much of the functionality is already there, hence why I am suggesting extending Battery Notes. Thank you for considering this. I have been looking for an HA solution for this for years and was very excited to discover Battery Notes being extended similarly to how you could track other replaceables.

**Comment by andrew-codechimp** [comment]
I have thought about "Maintenance Notes" as a concept but it stretches what I want to do with this integration too far when you start to consider things like maintenance schedule calendars and brand new devices that don't exist in HA.

Battery Notes is a good starting point for anyone wanting to develop a new integration but at the moment I wouldn't have time to develop and maintain that.

I'll move this to ideas to keep it visible in case another integration dev would like to take up the challenge, it may also be worth raising it in the HA forums.
